File: 1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:

        lucky = []
        minn = set()
        for row in matrix:
            minn.add(min(row))

        col = list(zip(*matrix))
     
        for c in col:
            maxx = max(c)
            if maxx in minn:
                lucky.append(maxx)
            
        return lucky

File: 1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.go
func min(arr []int)(int, int) {
    minVal := arr[0]
    minIndex := 0
    for index, value := range arr {
        if value < minVal {
            minVal = value
            minIndex = index 
        }
    }
    return minVal , minIndex
}

func luckyNumbers (matrix [][]int) []int {

    var lucky []int
    
    var row int = len(matrix)

    for i:=0 ; i < row ; i++{
        minVal , minIndex := min(matrix[i])

        for j:=0 ; j < row; j++{
            if matrix[j][minIndex] > minVal{
                break
            }
            if j == len(matrix) - 1{
                lucky = append(lucky, minVal)
            }
        }
    }

    return lucky

}
